### PR TITLE
Return thenables for lazy evaluation instead of a promise

### LINKED ISSUE #8: Switch to lazy evaluation via getter instead of nextTick
Would eliminate the need for a special scheduling mechanism and avoid Bluebird reporting as unhandled

cc @blacksonic @gaastonsr
ref https://github.com/petkaantonov/bluebird/issues/313

### PR DESCRIPTION
**Pull request by bendrucker**
@blacksonic @gaastonsr

Would really appreciate it if you guys could take a look. I switched from returning a promise with deferred resolution to a thenable. Basically the new `thenable` function builds up an object that looks and works like a promise from the promise implementation in use. It won't actually create a promise until a method on the return value (e.g. `then` or `catch`) is invoked. 

I'd like to try to handle some edge cases a little better but I can definitely ship this in December.

Closes #8

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -3,23 +3,33 @@
...
-function schedule (fn) {
-  return function (resolve, reject) {
-    scheduler(function () {
-      fn(resolve, reject);
+function thenable (promiseFactory) {
+  return Object.getOwnPropertyNames(Promise.prototype)
+    .filter(function (method) {
+      return method !== 'then';
+    })
+    .reduce(function (acc, method) {
```
Comment (gaastonsr) [inline src/index.js]: `acc` variable name feels cryptic at first.

Comment (bendrucker) [inline src/index.js]: `acc` is short for accumulator and its a pretty common var name. Next time I make changes I'll think whether there's a clearer option.

Comment (gaastonsr) [inline src/index.js]: Now it makes sense. I knew I was missing something. Everything looks good, sorry for the delay.

Comment (bendrucker) [inline src/index.js]: Can't hurt to make the source a little clearer

5d8d128d69feb92651457d25ba56ba74cd166bd5

**Pull request merged**